lib/client/api.go

```diff
@@ -2724,13 +2766,75 @@ func (tc *TeleportClient) runCommandOnNodes(ctx context.Context, clt *ClusterCli
 			}
 			defer nodeClient.Close()
 
-			fmt.Printf("Running command on %v:\n", nodeName(address))
+			displayName := nodeName(node.addr)
+			if node.hostname != "" {
+				displayName = node.hostname
+			}
+			fmt.Printf("Running command on %v:\n", displayName)
 
-			return trace.Wrap(nodeClient.RunCommand(ctx, command))
+			if err := nodeClient.RunCommand(ctx, command); err != nil && tc.ExitStatus == 0 {
+				return trace.Wrap(err)
+			}
+			resultsCh <- execResult{
+				hostname:   displayName,
+				exitStatus: tc.ExitStatus,
+			}
+			return nil
 		})
 	}
 
-	return trace.Wrap(g.Wait())
+	if err := g.Wait(); err != nil {
```

Comment:
If there's a problem with one command goroutine we don't print out results, even though fx 9 out of 10 nodes might already succeeded 🤔